### DISCUSSION TITLE
Enable hyperspace in spark thriftserver

### DISCUSSION THREAD

**Discussion by ashokblend** [microsoft/hyperspace#499] [Q&A] [answered]
We have server code which connects to spark jdbc(thrift) server to execute sql.

We can have batch jobs which can create index for our data lake.

But we are not getting how to enable hyperspace for spark thrift server, where we do not have control in initialising sparksession.

So question is how to enable hyperspace in spark thriftserver ?

**Comment by sezruby** [comment, marked as answer]
For now, there's no way to enable hyperspace in spark thriftserver. 

One thing we can do is creating a SparkSessionExtension for Hyperspace, and support ENABLE/DISABLE HYPERSPACE sql or enable hyperspace for all sessions when the extension is added. 

For example, in Delta Lake: 
https://github.com/delta-io/delta/blob/master/core/src/main/scala/io/delta/sql/DeltaSparkSessionExtension.scala
`--conf "spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension" `

**Reply by sezruby** [reply]
@ashokblend We're going to implement a SparkSessionExtension for Hyperspace soon. I think it would be a good addition. Thanks! 😄
